### PR TITLE
Feature/GitHub action tag and publish docker image

### PR DESCRIPTION
**Pull request by ahmedelkashev**
### Description of the Change

1. Add a pipeline to build and push fuhon image to harbor.
2. Keep the DRYness of code by having a .src file for the pipeline that has non repeatable code parts, and a script that generates the workflow from the .src file, and a pre-commit to automate that process.
3. Fix the filcrypto installation script.
4. Add a dockerfile to build the fuhon image and include it in the pipeline.

### Usage Examples or Tests *[optional]*

1. Navigate to the .github/ directory
2. Run ‘./make-workflows —worktree’

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
+FROM ubuntu:20.04 AS base
```
Comment (turuslan) [inline Dockerfile:1]: Duplicate file `Dockerfile` and `docker/Dockerfile`.

File: .github/build-cpp-filecoin.src.yml [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    inputs:
+      build_spec:
+        description: 'See chatops-gen-matrix.sh, example "/build ubuntu macos gcc-9 burrow"'
+        required: false
+        default: '/build'
+  # issue_comment:
+  #   types: [created, edited]
+  schedule:
+    # Triggers the workflow every day at 5:30 and 17:30 UTC
+    - cron: '30 5,17 * * *'
```
Comment (turuslan) [inline .github/build-cpp-filecoin.src.yml:19]: Is it needed?  
`on.workflow_dispatch.inputs.build_spec` is not used.  
Do we need cron?

File: .github/build-cpp-filecoin.src.yml [hunk 3]
```diff
@@ -0,0 +1,204 @@
+name: Fuhon (cpp-Filecoin)
+
+on:
+  push:
+    branches: [master, feature/github-action-tag-and-publish-docker-image]
+    tags: '**'
+  pull_request:
+    branches: [master, feature/github-action-tag-and-publish-docker-image]  ## target branches
```
Comment (turuslan) [inline .github/build-cpp-filecoin.src.yml:8]: Will `feature/github-action-tag-and-publish-docker-image` be removed from these lists before merging into `master`?

Or will new commits be pushed to `feature/github-action-tag-and-publish-docker-image` after PR is merged?

Comment (ahmedelkashev) [reply .github/build-cpp-filecoin.src.yml:8]: the branch will be removed, it is there now just for testing. (will create a new PR specifically for it).

File: .github/build-cpp-filecoin.src.yml (hunk 2 above)
Comment (ahmedelkashev) [reply .github/build-cpp-filecoin.src.yml]: removed

**Review by turuslan — APPROVED** [review]

**Review by ortyomka — APPROVED** [review]

**Pull request merged**